### ISSUE TITLE
action hostsdeny throws ERROR in logfile but does the job

### ISSUE THREAD

**Issue by koeppea** [fail2ban/fail2ban#1293]
Hi

When I try to use the default command on a Debian 8  ( *however I had to explicitely install the package ed* ) , I always get an error when the *unban action* in the **hostsdeny.conf** is being invoked.
However, the line disappears in the end.
The following output is produced using a *bantime* of 60 and running `tail -f /var/log/fail2ban.log` in the background.
```
2016-01-07 23:21:08,176 fail2ban.actions[26268]: WARNING [ssh] Ban 192.168.1.1

root@localhost /etc/fail2ban # tail /etc/hosts.deny 
#
# The PARANOID wildcard matches any host whose name does not match its
# address.
#
# You may wish to enable this to ensure any programs that don't
# validate looked up hostnames still leave understandable logs. In past
# versions of Debian this has been the default.
# ALL: PARANOID

ALL: 192.168.1.1
root@localhost /etc/fail2ban # 
root@localhost /etc/fail2ban # 
root@localhost /etc/fail2ban # 2016-01-07 23:22:08,198 fail2ban.actions[26268]: WARNING [ssh] Unban 192.168.1.1
2016-01-07 23:22:08,230 fail2ban.actions.action[26268]: ERROR   echo -e "/^ALL: 192.168.1.1$/\nd\nw\nq" | ed /etc/hosts.deny returned 100

root@localhost /etc/fail2ban # 
root@localhost /etc/fail2ban # 
root@localhost /etc/fail2ban # tail /etc/hosts.deny 
# for further information.
#
# The PARANOID wildcard matches any host whose name does not match its
# address.
#
# You may wish to enable this to ensure any programs that don't
# validate looked up hostnames still leave understandable logs. In past
# versions of Debian this has been the default.
# ALL: PARANOID

root@localhost /etc/fail2ban # 

```

Why shows the logfile 
**ERROR   echo -e "/^ALL: 192.168.1.1$/\nd\nw\nq" | ed /etc/hosts.deny returned 100** ?

When I try to reproduce this by explicitely running these commands on the console, the return code is 0:
```

root@localhost /etc/fail2ban # 
root@localhost /etc/fail2ban # echo "ALL: 192.168.1.1" >> /etc/hosts.deny 
root@localhost /etc/fail2ban # 
root@localhost /etc/fail2ban # 
root@localhost /etc/fail2ban # tail /etc/hosts.deny 
#
# The PARANOID wildcard matches any host whose name does not match its
# address.
#
# You may wish to enable this to ensure any programs that don't
# validate looked up hostnames still leave understandable logs. In past
# versions of Debian this has been the default.
# ALL: PARANOID

ALL: 192.168.1.1
root@localhost /etc/fail2ban # 
root@localhost /etc/fail2ban # 
root@localhost /etc/fail2ban # echo -e "/^ALL: 192.168.1.1$/\nd\nw\nq" | ed /etc/hosts.deny
897
ALL: 192.168.1.1
880
root@localhost /etc/fail2ban # echo $?
0
root@localhost /etc/fail2ban # 
root@localhost /etc/fail2ban # 
root@localhost /etc/fail2ban # tail /etc/hosts.deny 
# for further information.
#
# The PARANOID wildcard matches any host whose name does not match its
# address.
#
# You may wish to enable this to ensure any programs that don't
# validate looked up hostnames still leave understandable logs. In past
# versions of Debian this has been the default.
# ALL: PARANOID

root@localhost /etc/fail2ban # 
```

**Comment by sebres** [comment]
> When I try to reproduce this by explicitely running these commands on the console, the return code is 0

Which fail2ban version? Old configs?
I believe we have already fixed it, because `echo` may not have the option `-e` in some shells;

In newest filter it was realized without `-e` - https://github.com/fail2ban/fail2ban/blob/master/config/action.d/hostsdeny.conf#L43

I will close the issue, please reopen if I'm wrong

**Comment by koeppea** [comment]
Hi, 
I'm using fail2ban from Debian's stable repositories. The version is ATM 0.8.13. However I realized that my modified *hostsdeny.conf* was still in place (yes blame me, meanwhile I read the man-pages and use my own .local files).
I removed it and restored the original *hostsdeny.conf* from the repositories and disabled my .local file.

I can confirm that you're right. I don't know why I came across this prior but using the version 0.8.13 *hostsdeny.conf*, it's working w/o the mentioned error message.

Thanks for pointing me into the right direction.
